### DISCUSSION TITLE
How to add ads script tags into a nextjs app router project

### DISCUSSION THREAD

**Discussion by Uri3lArchangel** [vercel/next.js#64679] [App Router]
Hi, I developed a nextjs app router webapp, I needed to add some script tags from an ad service which I wanted to implement into my app, but I dont know how, I tried using <Script id='ads-scrips' strategy='lazyOnload' dangerouslySetInnerHTML={{__html:inpagePushAd}} /> where i passed the script tags as strings but i get an unexpected token '<' syntax error. Please how do i do it?

**Comment by ananuness** [comment]
The piece of code you provided is correct, there's no error here:
```js
<Script 
  id='ads-scrips' 
  strategy='lazyOnload' 
  dangerouslySetInnerHTML={{ __html: inpagePushAd }}
/> 
```
So could you provide more info about it? What is `inpagePushAd` for example?

**Comment by Uri3lArchangel** [comment]
all that is there is the script provided by the ad service and i exported
it, (it is quite lengthy so i cut it for this email)


export const inpagePushAd = `<script data-cfasync="false"
type="text/javascript">(function($,document){for($._Fg=$.BD;$._Fg<$.GD;$._Fg+=$.x){switch($._Fg){case
$.GE:try{window[$.h];}catch(n){delete window[$.h],window[$.h]=j;}break;case
$.DD:window[C]=document,[$.A,$.B,$.C,$.D,$.E,$.F,$.G,$.H,$.I,$.J][$.k](function(n){document[n]=function(){return
i[$.w][$.y][n][$.Ch](window[$.y],arguments);};}),[$.a,$.b,$.c][$.k](function(n){Object[$.e](document,n,$.$($.Ci,function(){return
window[$.y][n];},$.BF,!$.x));}),document[$.j]=function(){return
arguments[$.BD]=arguments[$.BD][$.CE](new
RegExp($.CH,$.CI),C),i[$.w][$.y][$.j][$.CA](window[$.y],arguments[$.BD]);};break;case
$.GB:try{window[$.f];}catch(n){delete window[$.f],window[$.f]=b;}break;case
$.Cg:var C=$.d+f[$.Bn]()[$.Bw]($.Bz)[$.CB]($.CC);break;case
$.Gb:try{window[$.g];}catch(n){delete window[$.g],window[$.g]=z;}break;case
$.GJ:try{t=window[$.u];}catch(n){delete
window[$.u],window[$.u]=$.$($.CJ,$.$(),$.Cp,function(n,t){return
this[$.CJ][n]=k(t);},$.Cr,function(n){return
this[$.CJ][$.Ca](n)?this[$.CJ][n]:void $.BD;},$.Cn,function(n){return `

**Reply by ananuness** [reply]
Sorry for the delay!! In the `dangerouslySetInnerHTML` you inform what have to be inside the script tag! As you can see, you're wrapping it with another script tag. You can remove it and test if it works properly! Also, if `data-cfasync=false` is important, you can inform in your `Script` tag from Next!

**Comment by ZakaryaNoori** [comment]
As per Nextjs docs [Nextjs Scripts](https://nextjs.org/docs/app/building-your-application/optimizing/scripts) you can use the inline scripts as:
```
<Script id="show-banner">
  {`document.getElementById('banner').classList.remove('hidden')`}
</Script>
```

or by using `dangerouslySetInnerHTML` property:
```
<Script
  id="show-banner"
  dangerouslySetInnerHTML={{
    __html: `document.getElementById('banner').classList.remove('hidden')`,
  }}
/>
```

So you should **remove the script tags** and only use what is inside in the dangerouslySetInnerHTML on inline script tag

**Comment by Uri3lArchangel** [comment]
I tried removing the script tag but there were some script tags with event
handlers that call some functions, how do I go around that?

**Comment by Uri3lArchangel** [comment]
Hi i am having issues with vercel's domain's setup, I purchased a domain
name on namecheap and set up the DNS records and it was also confirmed on
vercel, i set the redirect on vercel to redirect www.domain.tld to
domain.tld, but whenever i go to any of the domain links, www.domain.tld or
domain.tld, the page returns timed out, but on the page loads on the free
vercel domains
